src/consent-manager/dialog.tsx

```diff
@@ -122,22 +122,34 @@ type Props = {
   title: React.ReactNode
   buttons: React.ReactNode
   width?: string
-};
-
-const container = document.createElement('div');
+}
 
-const Dialog: FC<Props> = ({ onCancel = undefined, onSubmit, title, innerRef, children, buttons, width = '960px' }) => {
-  const titleId = nanoid();
-  const rootRef = useRef<HTMLDListElement>(null);
-  let form: HTMLFormElement;
+const Dialog: FC<Props> = ({
+  onCancel = undefined,
+  onSubmit,
+  title,
+  innerRef,
+  children,
+  buttons,
+  width = '960px',
+}) => {
+  let container: HTMLDivElement | undefined
+  const titleId = nanoid()
+  const rootRef = useRef<HTMLDListElement>(null)
+  let form: HTMLFormElement
 
+  useEffect(() => {
+    container = document.createElement('div')
```

Comment:
if you just do like that, you will not trigger a new render (maybe it works, but it's just luck),
but if put this line in a useEffect, I suppose you want to trigger a new render when you set set the container

so you have to do something like 
cons [container, setContainer] = useState<HTMLDivElement | undefined>(undefined)

useEffect(() => {
    setContainer(document.createElement('div'))
}, [])